sound/soc/sof/intel/hda-dai.c

```diff
@@ -451,5 +468,33 @@ struct snd_soc_dai_driver skl_dai[] = {
 	.name = "Alt Analog CPU DAI",
 	.ops = &hda_link_dai_ops,
 },
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_DEBUG_PROBES)
+{
+	.name = "Probe Injection0 CPU DAI",
+	.compress_new = snd_soc_new_compress,
+	.cops = &sof_probe_compr_ops,
+	.playback = {
+		.stream_name = "Probe Injection0",
+		.channels_min = 1,
+		.channels_max = 8,
+		.rates = SNDRV_PCM_RATE_48000,
+		.rate_min = 48000,
+		.rate_max = 48000,
+	},
+},
+{
+	.name = "Probe Extraction CPU DAI",
+	.compress_new = snd_soc_new_compress,
+	.cops = &sof_probe_compr_ops,
+	.capture = {
+		.stream_name = "Probe Extraction",
+		.channels_min = 1,
+		.channels_max = 8,
+		.rates = SNDRV_PCM_RATE_48000,
+		.rate_min = 48000,
+		.rate_max = 48000,
+	},
+},
+#endif
```

Comment:
I think we should create those DAIs dynamically instead of with a table, and make the number of injectors a Kconfig/kernel module parameter.

Reply:
Tried today assignment of member 'drv' (struct sof_dev_desc::ops) stopped me. Anyway, I'd refrain from expanding on this for current PR. That can be done later as an improvement.
95% cases are covered with single extraction point. 99% with multiple extraction point support. 99,(9)% if you add single injection on top of that. None ever seen multiple injectors being engaged throughout the entire lifetime of cAVS.
That is not an excuse of course, but an argument for moving this topic to an improvement section.